### PR TITLE
[TEMPLATES] added titlecard plugin

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3b6793c55b54173/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3b6793c55b54173/output.txt

Total script time: 0.38 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/04497828e054cba/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/04497828e054cba/output.txt

Total script time: 0.14 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/411d38130508294/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/411d38130508294/output.txt

Total script time: 0.15 mins

File: templates/supported/shared/plugins/titlecard/popcorn.titlecard.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    style.bottom = 0;
+
+    media.parentNode.appendChild( ctxContainer );
+    return ctxContainer;
+  }
+
+  Popcorn.plugin( "titlecard", {
+    manifest: {
+      about: {
+      name: "Popcorn titlecard Plugin",
```
Comment (mjschranz) [inline templates/supported/shared/plugins/titlecard/popcorn.titlecard.js]: indent 2 spaces

File: templates/supported/shared/plugins/titlecard/popcorn.titlecard.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      version: "0.1",
```
Comment (mjschranz) [inline templates/supported/shared/plugins/titlecard/popcorn.titlecard.js]: ^

File: templates/supported/shared/plugins/titlecard/popcorn.titlecard.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      author: "@k88hudson"
```
Comment (mjschranz) [inline templates/supported/shared/plugins/titlecard/popcorn.titlecard.js]: ^

File: templates/supported/shared/plugins/titlecard/popcorn.titlecard.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        type: "text",
+        label: "title",
+        "default": "Popcorn rules dude!"
+      },
+      subheading: {
+        elem: "input",
+        type: "text",
+        label: "Subheading",
+        "default": "Popcorn.js"
+      }
```
Comment (mjschranz) [inline templates/supported/shared/plugins/titlecard/popcorn.titlecard.js:56]: Indent all the options in 2 spaces from start to subheading so it's looks something like....

<pre>
options: {
  start: {
    elem: "input",
    type: "text",
    label: "In"
  },
  end: {
    elem: "input",
    type: "text",
    label: "Out"
  },
  title: {
    elem: "input",
    type: "text",
    label: "title",
    "default": "Popcorn rules dude!"
  },
  subheading: {
    elem: "input",
    type: "text",
    label: "Subheading",
    "default": "Popcorn.js"
  }
}
</pre>

File: templates/supported/shared/plugins/titlecard/popcorn.titlecard.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  /**
+   * titlecard Popcorn plug-in
+   * Based on text plugin by @humph
+   *
+   **/
+
+  // Assumes the video is wrapped in a container.
+  function createContainer( context, id ) {
+
+  var ctxContainer = context.container = document.createElement( "div" ),
```
Comment (mjschranz) [inline templates/supported/shared/plugins/titlecard/popcorn.titlecard.js]: indent 2 spaces

**Pull request merged**